brave/browser-laptop
# Refactor addEditBookmarkHanger.js with commonForm

Closes #6040
Addresses #7989

- Added `CommonFormBookmarkHanger` to commonForm.js
- Added `globalStyles.spacing.bookmarkHangerMaxWidth` to global.js
  Max-width of the bookmark hanger was set to 350px. Also, setting white-space:normal makes it possible for lines to be wrapped.

- Applied `Dialog` to addEditBookmarkHanger.js instead of addEditBookmark.js
- Added cancel button with `this.onClose` as 'fa fa-close' was removed

- Added const `arrowUp` and `dialogHanger`
- Added `styles.bookmarkHanger` which is applied if `!this.props.isModal`
- Copied `CommonFormSection` and `CommonFormTitle` from commonForm.js, adding `CommonFormBookmarkHangerTitle` to addEditBookmarkHanger.js
  The margin-top of the title should be reduced if `this.props.isModal`.

- Changed backgroundColor of commonForm to `#fff`

- Add zindex to elements on navigation bar to let tests pass
  The value was set to higher than that of dialog, which is 3000.

- Updated test code

Also:
- Replaced divs with sections
- Removed styles from form.less
- Removed `wideButton` from buttons
- Added testIds to buttons

Auditors:

Test Plan 1:
1. Open about:preferences
2. Disable Home button
3. Click the star icon on the URL bar
4. Make sure the arrow up appears exactly under the star
5. Open about:preferences
6. Enable Home button
7. Click the star icon again
8. Make sure th arrow up appears exactly under the star

Test Plan 2:
1. Open about:bookmarks
2. Click the star icon on the URL bar
3. Click "Done"
4. Click the star icon again
5. Click "Remove"
6. Make sure the bookmark is successfully removed

Test Plan 3:
1. Click "Add Folder" icon on about:bookmarks
2. Click "Cancel"
3. Make sure the dialog is closed

Test Plan 4:
1. Create a bookmark folder on about:bookmarks
2. Edit the folder via context menu on about:bookmarks
3. Click "Remove"
4. Make sure the folder is successfully removed

Test Plan 5:
1. Click the star icon on about:bookmarks
2. Click "Cancel"
3. Make sure the dialog is closed
4. Click the star icon again
5. Create a bookmark
6. Edit the bookmark via context menu
7. Click "Remove"
8. Make sure the bookmark is successfully removed


Test Plan:
